rs_server/CADIP/api/cadu_download.py

```diff
@@ -3,283 +3,228 @@
+            status.in_progress(db)  # updates the database
+            # notify the main thread that the download will be started
+            thread_started.set()
+            init = datetime.now()
+            data_retriever.download(cadu_id, name)
+            logger.info(
+                "%s : %s : File: %s downloaded in %s",
                 os.getpid(),
                 threading.get_ident(),
-                end,
-                eop.location,
-                end - init,
-            ),
-        )
-    except Exception as e:
-        print("{} : {} : {}: Exception caught: {}".format(os.getpid(), threading.get_ident(), datetime.now(), e))
-        update_db(id, "failed")
-        return
-
-    if obs is not None and len(obs) > 0:
-        # TODO: the secrets should be set through env vars
-        secrets = {
-            "s3endpoint": None,
-            "accesskey": None,
-            "secretkey": None,
-        }
-        S3StorageHandler.get_secrets(secrets, "/home/" + os.environ["USER"] + "/.s3cfg")
-        print(f"secrets = {secrets}")
-        s3_handler = S3StorageHandler(secrets["accesskey"], secrets["secretkey"], secrets["s3endpoint"], "sbg")
-
-        filename = uri_to_path(eop.location)
-        obs_array = obs.split("/")
-        print(
-            "filename = {} | obs_array = {} | join = {} | filename {}".format(
-                filename,
-                obs_array,
-                "/".join(obs_array[2:]),
-                "/".join(obs_array[2:]) + name,
-            ),
-        )
-
-        # TODO check the length
-        s3_config = PrefectPutFilesToS3Config(s3_handler, [filename], obs_array[2], "/".join(obs_array[3:]), 0)
-        asyncio.run(prefect_put_files_to_s3.fn(s3_config))
-
-        os.remove(filename)
-
-    update_db(id, "succeeded")
+                name,
+                datetime.now() - init,
+            )
+        except Exception as exception:
+            logger.error(
+                "%s : %s : %s: Exception caught: %s",
+                os.getpid(),
+                threading.get_ident(),
+                datetime.now(),
+                exception,
+            )
+            update_db(db, status, EDownloadStatus.FAILED, repr(exception))
+            return
+
+        if obs is not None and len(obs) > 0:
+            try:
+                # NOTE: The environment variables have to be set from outside
+                # otherwise the connection with s3 endpoint fails
+                secrets = {
+                    "s3endpoint": os.environ["S3_ENDPOINT"],
+                    "accesskey": os.environ["S3_ACCESSKEY"],
+                    "secretkey": os.environ["S3_SECRETKEY"],
+                }
+                s3_handler = S3StorageHandler(secrets["accesskey"], secrets["secretkey"], secrets["s3endpoint"], "sbg")
+                obs_array = obs.split("/")                
+                s3_config = PrefectPutFilesToS3Config(
```

Comment:
Maybe we can create a small function that parse and transform content of <<obs>> param, since using slicing directly in a function call is hard to read / maintain.